src/backend/luna.rs

```diff
@@ -0,0 +1,100 @@
+use rusb::{
+    GlobalContext, DeviceHandle,
+};
+use std::collections::VecDeque;
+use std::thread;
+use std::sync::mpsc::{channel, Receiver};
+use std::time::Duration;
+
+const VID: u16 = 0x1d50;
+const PID: u16 = 0x615b;
+
+const ENDPOINT: u8 = 0x81;
+
+const READ_LEN: usize = 0x4000;
+
+pub struct LunaDevice {
+    handle: Option<DeviceHandle<GlobalContext>>,
+    worker: Option<thread::JoinHandle<DeviceHandle<GlobalContext>>>,
+    receiver: Option<Receiver<Vec<u8>>>,
+}
```

Comment:
Rather than having all these be `Option`, and potentially `None` if you have the thing in the wrong state, could we split this structure up?

I'm thinking that `start()` could return a `LunaCapture` which would own the worker thread and the receiver, and then `next()` could be on the `LunaCapture`, which would avoid the need to do `self.receiver.as_ref().unwrap()` in there.